Decal Registry improvements

bigkahuna443 opened the pull request:
Some improvements to Decal Registry:
1. Improved [wiki page](https://github.com/EverestAPI/Resources/wiki/Decal-Registry)
2. Made property syntax more consistent
3. Added assembly/mod name logging for property name conflicts (previously would not tell you which mod was replacing the property when the conflict occurred)
4. Removed unnecessary logic for `overlay`
5. Made `scale` compatible with other tags

microlith57 commented on Celeste.Mod.mm/Mod/Registry/DecalRegistry.cs:
```diff
@@ -37,82 +38,50 @@ public static class DecalRegistry {
+                float height = attrs["height"] != null ? float.Parse(attrs["height"].Value) : 16f;
+                int index = attrs["index"] != null ? int.Parse(attrs["index"].Value) : SurfaceIndex.ResortRoof;
                 bool blockWaterfalls = attrs["blockWaterfalls"] != null ? bool.Parse(attrs["blockWaterfalls"].Value) : true;
                 ((patch_Decal)decal).MakeSolid(x, y, width, height, index, blockWaterfalls);
```
currently `MakeSolid` creates a solid with `safe: true`; is this something that might benefit from configurability (for example if a decal is a solid staticmover?)

bigkahuna443 replied:
I don't think anyone's requested this but probably wouldn't hurt to add. Maybe in a different PR since we discussed other changes to the tag anyway

microlith57 replied:
might as well put it in my decal registry parity pr

microlith57 replied:
now in https://github.com/EverestAPI/Everest/pull/463

microlith57 reviewed (approved):
looks good!

bigkahuna443 committed:
When a property name conflict occurs, include the name of the assembly/mod that replaces the property to the log.

For the style changes, I didn't touch everything, and maybe a wider style discussion needs to happen, but felt like property definitions would benefit from a consistent format.

bigkahuna443 committed:
Remove unneeded 'this'

bigkahuna443 committed:
Apply scaling to other properties

Applies "scale" as the first property, and then adjusts offsets, radii, etc. by the scale for compatibility. Without this, the scale attribute breaks pretty much all of these properties.

These problems existed before with Ahorn/Lonn scaling, but this feels a bit more urgent now that there's an Everest property for it.
